knowrob_kautham/prolog/knowrob_kautham.pl

```diff
@@ -0,0 +1,81 @@
+
+  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+  ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+  WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+  DISCLAIMED. IN NO EVENT SHALL <COPYRIGHT HOLDER> BE LIABLE FOR ANY
+  DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+  (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+  ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+  (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+  SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+@author Daniel Beßler
+@license BSD
+*/
+:- module(knowrob_kautham,
+    [
+        comp_affordanceocclusion/2
+    ]).
+
+
+:- use_foreign_library('libkauthamwrapper.so').
+
+:- use_module(library('semweb/rdf_db')).
+:- use_module(library('semweb/rdfs')).
+:- use_module(library('semweb/owl')).
+:- use_module(library('semweb/owl_parser')).
+:- use_module(library('knowrob/owl')).
+:- use_module(library('knowrob/beliefstate')).
+:- use_module(library('knowrob/transforms')).
+:- use_module(library('knowrob_assembly')).
+
+:- rdf_db:rdf_register_ns(knowrob, 'http://knowrob.org/kb/knowrob.owl#', [keep(true)]).
+:- rdf_db:rdf_register_ns(knowrob_assembly, 'http://knowrob.org/kb/knowrob_assembly.owl#', [keep(true)]).
+:- rdf_db:rdf_register_ns(srdl2comp, 'http://knowrob.org/kb/srdl2-comp.owl#', [keep(true)]).
+:- rdf_db:rdf_register_ns(knowrob_paramserver, 'http://knowrob.org/kb/knowrob_paramserver.owl#', [keep(true)]).
+
+:-  rdf_meta
+  comp_affordanceocclusion(r, ?).
+
+
+%% comp_affordanceocclusion(+Affordance, ?Part)
+%
+comp_affordanceocclusion(Part, GraspingAffordance) :-
+% Retrieve transform associated to GraspingAffordance
+  rdf_has(GraspingAffordance, knowrob_assembly:'graspAt', GraspSpecification),
+  rdf_has(GraspSpecification, knowrob_paramserver:'hasGraspTransform', GraspTransform),
+  rdf_has(GraspTransform, knowrob_paramserver:'hasTranslation', GraspTranslation),
+  rdf_has(GraspTransform, knowrob_paramserver:'hasRotation', GraspRotation),
+  rdf_has(GraspTranslation, knowrob_paramserver:'hasX', GTx),
+  rdf_has(GraspTranslation, knowrob_paramserver:'hasY', GTy),
+  rdf_has(GraspTranslation, knowrob_paramserver:'hasZ', GTz),
+  rdf_has(GraspRotation, knowrob_paramserver:'hasX', GRx),
+  rdf_has(GraspRotation, knowrob_paramserver:'hasY', GRy),
+  rdf_has(GraspRotation, knowrob_paramserver:'hasZ', GRz),
+  rdf_has(GraspRotation, knowrob_paramserver:'hasW', GRw),
+% Retrieve TargetPart (associated to GraspingAffordance)
+  rdf_has(TargetPart, knowrob_assembly:'hasAffordance', GraspingAffordance),
+% Retrieve pose associated to TargetPart
+  belief_at(TargetPart, [_, _, [OTx, OTy, OTz], [ORx, ORy, ORz, ORw]]),
```

Comment:
you should provide map frame here if you want to get the pose in world coordinates.

Reply:
When checking what belief_at does, it seems it implicitly sets the frame to return the coordinates in to world.

Reply:
Sure about this? I thought that it provides it then as is. Would make sense to avoid transformation if not necessary.
There is a predicate `map_frame_name` that you can use to avoid hardcoded "map" in the code